File: samples/generators/jpa/src/main/javagen/topmodel/jpa/sample/demo/dtos/securite/profil/ProfilItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package topmodel.jpa.sample.demo.dtos.securite.profil;

import jakarta.annotation.Generated;

/**
 * Détail d'un profil en liste.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface ProfilItem {

	/**
	 * Id technique.
	 */
	Integer getId();

	/**
	 * Libellé du profil.
	 */
	String getLibelle();

	/**
	 * Nombre d'utilisateurs affectés au profil.
	 */
	Long getNombreUtilisateurs();

	/**
	 * Hydrate values of instance.
	 * @param id value to set.
	 * @param libelle value to set.
	 * @param nombreUtilisateurs value to set.
	 */
	void hydrate(Integer id, String libelle, Long nombreUtilisateurs);
}

File: samples/generators/angular/src/appgenerated/model/securite/profil/profil-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {DO_ENTIER, DO_ID, DO_LIBELLE} from "@domains";

export interface ProfilItem {
    id?: number,
    libelle?: string,
    nombreUtilisateurs?: number
}

export const ProfilItemEntity = {
    id: {
        type: "field",
        name: "id",
        domain: DO_ID,
        isRequired: false,
        label: "securite.profil.profil.id"
    },
    libelle: {
        type: "field",
        name: "libelle",
        domain: DO_LIBELLE,
        isRequired: true,
        label: "securite.profil.profil.libelle"
    },
    nombreUtilisateurs: {
        type: "field",
        name: "nombreUtilisateurs",
        domain: DO_ENTIER,
        isRequired: true,
        label: "securite.profil.profilItem.nombreUtilisateurs"
    }
} as const
